door-runtime/src/jsMain/kotlin/wrappers/SQLiteDatasourceJs.kt

```diff
@@ -20,13 +21,29 @@ class SQLiteDatasourceJs(private val dbName: String, private val worker: Worker)
 
     private val pendingMessages = mutableMapOf<Int, CompletableDeferred<WorkerResult>>()
 
+    private val executedSqlQueries = mutableMapOf<Int, String>()
+
+    var generatedKeys: ResultSet = SQLiteResultSet(arrayOf())
+
     init {
-        worker.onmessage = { it: dynamic ->
-            val pendingCompletable = pendingMessages.remove(it.data["id"].toString().toInt())
+        worker.onmessage = { dbEvent: dynamic ->
+            val actionId = dbEvent.data["id"].toString().toInt()
+            val executedQuery = executedSqlQueries[actionId]
+            if(dbEvent.data["error"] != js("undefined")){
+                throw SQLException(dbEvent.data["error"].toString(),
```

Comment:
You need to get the pending completeable, and then use completeExceptionally(exception).